injection/sharedmain/main.go

```diff
@@ -76,9 +80,15 @@ func GetLoggingConfig(ctx context.Context) (*logging.Config, error) {
 	if err != nil {
 		return nil, err
 	}
+
 	return logging.NewConfigFromConfigMap(loggingConfigMap)
 }
 
+// LookupConfigMap retrieves a configmap with the given name
+func LookupConfigMap(ctx context.Context, name string) (*corev1.ConfigMap, error) {
+	return kubeclient.Get(ctx).CoreV1().ConfigMaps(system.Namespace()).Get(name, metav1.GetOptions{})
+}
```

Comment:
So you actually don't need to do this.  The configmap watcher will make sure that every watched configmap has its callback called.

Reply:
I think with the changes to `readProfilingFlag` to be context-insensitive this would be fine with this logic simply removed.

Reply:
Hmm. So this is getting really frustrating. I think the previous solution was completely same as you're suggesting now. So I'd basically need to revert the latest changes.

Reply:
Sorry, but the difference from what you had before is subtle.  Previously you had an Update function that *made no change* when the configmap value was unspecified [here](https://github.com/knative/pkg/pull/562#diff-89a311b4537c35ce0cb1f320c8e58237R81).

The way you have refactored this so that `readProfilingFlag` is separate and returns `false` in this case, so the problem is resolved and you shouldn't need all the extra logic.